### PR TITLE
Fix issue with tips for truncated text when switching pages in a table

### PR DESCRIPTION
**Pull request by vbanda28**
#### What does this PR do?
Resolve issue described here with truncated text: #6659. The issue occurs because tips for truncated text are not being updated when switching pages in a table. Resolved by dispatching an event "paginationupdate" in pagination.js when page is changed and adding an event listener in Text.js to call updateTip.

#### Where should the reviewer start?
Start with file pagination.js

#### What testing has been done on this PR?
Tested manually and ran test suite.

#### How should this be manually tested?
We can use the storybook component Visualization -> DataTable -> Pagination  for testing. The data in pagination.tsx can be updated with truncated text. The pagination.tsx file with local changes I made for testing can be found here: https://gist.github.com/vbanda28/84db2a7872a4183a0ee9dce031c96164

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?
#6659

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
Yes

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

File: src/js/components/Pagination/Pagination.js [hunk 1]
```diff
@@ -49,7 +49,11 @@ const Pagination = forwardRef(
       Math.min(page, totalPages) || 1,
     );
 
-    useEffect(() => setActivePage(page), [page]);
+    useEffect(() => {
+      setActivePage(page);
+      const pageEvent = new Event('paginationupdate');
```
Comment (taysea) [inline src/js/components/Pagination/Pagination.js:54]: Small naming comment, wondering if calling the event "pagechange" would be more descriptive.

**Review by taysea — COMMENTED** [review]
Functionality looks great, just one small naming comment.

**Review by taysea — APPROVED** [review]
LGTM. Thanks for the contribution!

**Review by jcfilben — APPROVED** [review]
Looks good!

**Pull request merged**